squad/ci/backend/lava.py

```diff
@@ -314,10 +318,16 @@ def __parse_results__(self, data, test_job):
                 if handle_lava_suite or result['suite'] != 'lava':
                     suite = result['suite'].split("_", 1)[-1]
                     res_name = "%s/%s" % (suite, result['name'])
+                    res_log = None
+                    if 'log_start_line' in result.keys() and \
+                            'log_end_line' in result.keys() and \
+                            result['log_start_line'] is not None and \
+                            result['log_end_line'] is not None:
+                        res_log = self.__download_test_log__(test_job.job_id, result['log_start_line'], result['log_end_line'])
```

Comment:
it looks like this will make an HTTP request for every single test result? we should probably download the full log once, and do this separation locally instead

Reply:
This will be tricky. Any hint how to retrieve only certain lines from the local file? Lava stores offsets to read from the file.

Reply:
I mean retrieve only certain lines without loading the whole file to memory.

Reply:
if you `open()` the file, it won't be read into memory as a whole unless you say so. if you have the offsets, you can `seek()` to them; otherwise you can read line by line with `readline()`